pkg/catalogv2/system/system.go

```diff
@@ -399,41 +406,47 @@ func (m *Manager) isInstalled(namespace, name, version, minVersion string, desir
 			return false, "", nil, err
 		}
 
-		desiredValue = map[string]interface{}{}
-		if err := json.Unmarshal(patchedJSON, &desiredValue); err != nil {
+		current, err := semver.NewVersion(r.Chart.Metadata.Version)
+		if err != nil {
 			return false, "", nil, err
 		}
 
-		ver, err := semver.NewVersion(release.Chart.Metadata.Version)
+		desired, err := semver.NewVersion(latestVersion)
 		if err != nil {
 			return false, "", nil, err
 		}
 
+		if minVersion == "" {
+			// Use exact version.
+			if !current.Equal(desired) {
```

Comment:
This logic relies on having exactly one release. If there are other releases, it's possible that there could be an existing release which matches this version exactly but you didn't process that one first, so you returned false and requested a new install. 

Currently, no part of the code makes that assertion. If you can elaborate on your confidence level here (or why you think that this won't return more than one release on a list) that would be helpful.

Reply:
1. This is the same behavior as the previous version if minVersion equals empty string. ( I wrote some additional tests to verify this.)
2. This logic relies on only one version of a release being deployed at a time which is enforced by Helm. The loop is to handle skipping over releases that are not deployed.